bin/configurer/utils/generate-local-user-mappings.sh

```diff
@@ -4,22 +4,50 @@ set -eu
 
 dir=$(dirname ${0})
 root_dir=$(realpath ${dir}/../../..)
-resources_dir=${root_dir}/service/src/main/resources
 
-function get_user_ids() {
-  docker run -e PGPASSWORD='openidm' --rm --network ccd-network postgres:11-alpine psql --host shared-db  --username openidm --tuples-only  --command "SELECT string_agg(fullobject->>'_id',',') FROM managedObjects WHERE fullobject->>'userName' LIKE '%$1%';" openidm
+mock_file=${root_dir}/docker/wiremock/__files/mockResponse.json
+mock_tmp_file=${root_dir}/docker/wiremock/__files/mockResponse.tmp.json
+users_file=${root_dir}/bin/configurer/users.json
+users_ids_tmp_file=${dir}/userIds.json.tmp
+
+user_mappings_file=${root_dir}/service/src/main/resources/application-user-mappings.yaml
+
+function query_db() {
+  docker run -e PGPASSWORD='openidm' --rm --network ccd-network postgres:11-alpine psql --host shared-db  --username openidm --tuples-only  --command "$1" openidm
+}
+
+function get_users_email_id_mappings() {
+  query_db "SELECT json_object(array_agg(fullobject->>'mail'), array_agg(fullobject->>'_id')) FROM managedObjects WHERE fullobject->>'mail' IS NOT NULL AND fullobject->>'userName' LIKE '%$1%';"
+}
+
+function get_users_ids() {
+  query_db "SELECT string_agg(fullobject->>'_id',',') FROM managedObjects WHERE fullobject->>'userName' LIKE '%$1%';"
+}
+
+function create_mock_response() {
+  jq --arg organisation $1 '[.[] | select(.email | contains($organisation))]' $users_file > $mock_file
+  echo $(get_users_email_id_mappings $1) > $users_ids_tmp_file
+
+  for i in `seq 1 $(jq '. | length' $mock_file)`
+  do
+    email=$(jq -r --argjson index $i '.[$index-1].email' $mock_file)
+    user_id=$(jq -r --arg email $email '.[$email]' $users_ids_tmp_file)
+    jq -r --argjson index $i --arg user_id $user_id '.[$index-1].userIdentifier=$user_id | .[$index-1].firstName=.[$index-1].email | .[$index-1].idamStatus="ACTIVE" | .[$index-1].idamStatusCode="0" | .[$index-1].idamMessage="" | .[$index-1].roles|=split(",")' $mock_file > $mock_tmp_file && mv $mock_tmp_file $mock_file
```

Comment:
Not a big fan of this solution. First it is not clear what the end result is cause of all these pipes. There is very strong coupling between input list of users ('users.json') and output mock response. That makes it less readable and unnecessarily more fragile.

Reply:
Added template file used to fill mock response and therefore reduced number of pipes on this line. But this needs to be strongly coupled with _users.json_ as the idea was to have single place for users definition that's used in all user-related scripts.

Reply:
I understand need for having source list of users, what I do not like is that that list, that is used in two or three different places is very strongly coupled with structure of the API that we mock. In that sense we have no freedom of adding anything to elements of the user list because if we do we will smuggle extra stuff to mocked responses. Same way unlikely change in PRD API structure might force us make change in two or three places that are completely unrelated.

In ideal world list of users should have its own structure and shell script should make that structure to structure of PRD API.

Cause of tight deadline I am not asking more for changes. Happy to accept as is ;)

Reply:
Here is 🥇 for you @pd-dudkiewicz for becoming our JQ expert :)